spec/services/assessors/assessment_proceeding_type_assessor_spec.rb

```diff
@@ -18,6 +18,60 @@ module Assessors
 
     describe ".call" do
       describe "successful result" do
+        context "asylum supported applicant" do
+          before { assessment.applicant.update!(receives_asylum_support: true) }
+
+          context "when using non-immigration/asylum proceeding type codes" do
+            # specify assessment results in order: gross_income_eligibility, disposable_income_eligibility, capital_eligibility
+            [
+              [:e, :cr, :e, "contribution_required"],
+              [:e, :cr, :i, "ineligible"],
+              [:e, :e, :cr, "contribution_required"],
+              [:e, :e, :e, "eligible"],
+              [:e, :e, :i, "ineligible"],
+              [:e, :i, :cr, "ineligible"],
+              [:e, :i, :e, "ineligible"],
+              [:e, :i, :i, "ineligible"],
+              [:e, :i, :p, "ineligible"],
+              [:i, :cr, :cr, "ineligible"],
+              [:i, :cr, :e, "ineligible"],
+              [:i, :cr, :i, "ineligible"],
+              [:i, :cr, :p, "ineligible"],
+              [:i, :e, :cr, "ineligible"],
+              [:i, :e, :e, "ineligible"],
+              [:i, :e, :i, "ineligible"],
+              [:i, :e, :p, "ineligible"],
+              [:i, :i, :cr, "ineligible"],
+              [:i, :i, :e, "ineligible"],
+              [:i, :i, :i, "ineligible"],
+              [:i, :i, :p, "ineligible"],
+              [:i, :p, :cr, "ineligible"],
+              [:i, :p, :e, "ineligible"],
+              [:i, :p, :i, "ineligible"],
+              [:i, :p, :p, "ineligible"],
+            ].each do |params|
+              it "updates the assessment eligibility record with the correct result" do
+                gross, disposable, capital, expected_result = params
+                expect(setup_and_test_result(gross, disposable, capital)).to eq(expected_result), "Expected #{gross}, #{disposable}, #{capital} to give #{expected_result}"
+              end
+            end
+          end
+
+          context "when using immigration/asylum proceeding type codes" do
+            let(:ptc) { "IM030" }
+
+            before do
+              assessment.proceeding_types.order(:ccms_code).first.update!(ccms_code: "IM030")
+              assessment.eligibilities.order(:proceeding_type_code).first.update!(proceeding_type_code: "IM030")
+            end
```

Comment:
I'm not a fan of this pattern, and in this case it looks like the test and code are tightly coupled. Could we not break this out into a setup with lets (or separate setups)?